gravitee-am-test/specs/gateway/mfa/mfa-enrollment-required.spec.ts

```diff
@@ -0,0 +1,226 @@
+        password: 'adminadmin',
+    },
+    domain: {
+        domainHrid: 'mfa-test-domain-enrollment-required',
+    },
+} as MfaTestContext;
+
+beforeAll(async () => {
+    await init(testContext);
+});
+
+afterAll(async () => {
+    await removeDomain(testContext);
+});
+
+beforeEach(async () => {
+    testContext.application = defaultClientSettings;
+    testContext.application.id = await createApp(testContext);
+    testContext.oidc.clientAuthorizationEndpoint = `${testContext.oidc.authorizationEndpoint}?response_type=code&client_id=${testContext.application.clientId}&redirect_uri=https://auth-nightly.gravitee.io/myApp/callback`;
+    testContext.domain.user = await createTestUser(testContext);
+    await logoutUser(testContext);
+});
+
+afterEach(async () => {
+    await deleteApplication(testContext.domain.domainId, testContext.admin.accessToken, testContext.application.id);
+});
+
+describe('When Enrollment REQUIRED ', () => {
+    it('and factor is not enrolled should enroll', async () => {
+        await patchApplication(
+            testContext.domain.domainId,
+            testContext.admin.accessToken,
+            defaultApplicationSettings(),
+            testContext.application.id,
+        );
+
+        const authResponse = await withRetry(() => get(testContext.oidc.clientAuthorizationEndpoint, 302));
+        const loginPage = await followUpGet(authResponse, 200);
+
+        let xsrf = extractDomValue(loginPage, '[name=X-XSRF-TOKEN]');
+        let action = extractDomAttr(loginPage, 'form', 'action');
+
+        const loginPostResponse = await withRetry(() => postForm(
+            action,
+            {
+                'X-XSRF-TOKEN': xsrf,
+                username: testContext.domain.user.username,
+                password: testContext.domain.user.password,
+                client_id: testContext.application.clientId,
+            },
+            {
+                Cookie: loginPage.headers['set-cookie'],
+                'Content-type': 'application/x-www-form-urlencoded',
+            },
+            302,
+            "/authorize"));
+
+        const enrollLocationResponse = await followUpGet(loginPostResponse, 302);
+        expect(enrollLocationResponse.headers['location']).toBeDefined();
+        expect(enrollLocationResponse.headers['location']).toContain('/enroll');
+    })
+    it.skip('and factor is enrolled but challenge is disabled should stop MFA flow', async () => {
+        await patchApplication(
+            testContext.domain.domainId,
+            testContext.admin.accessToken,
+            defaultApplicationSettings(),
+            testContext.application.id,
+        );
+
+        const cookie = await processMfaEnrollment();
+
+        const authResponse = await withRetry(() => get(testContext.oidc.clientAuthorizationEndpoint, 302, {Cookie: cookie.cookie}));
+
+        expect(authResponse.headers['location']).toBeDefined();
+        expect(authResponse.headers['location']).toContain('/challenge');
```

Comment:
based on the test description, should be redirect to the callback with a code IMO. wdyt?